### DISCUSSION TITLE
Adapter regelt nicht?

### DISCUSSION THREAD

**Discussion by FChristian** [MK-2001/ioBroker.go-e#124] [Q&A] [answered]
Hi, erst einmal dickes DANKE für den Adapter und die viele Arbeit, die da offensichtlich reingeflossen ist. Toll!

Meine Frage wäre: ich verstehe die unteren Teile in der Adapter Config so, dass man dort die PV-Leistung und den Hausverbrauch einträgt (habe mich an https://github.com/MK-2001/ioBroker.go-e/discussions/21#discussioncomment-888287 gehalten). Die eingetragenen Datenpunkte enthalten auch valide Werte, die alle 10sec aktualisiert werden. Da ich keine Batterie habe, habe ich die untersten beiden Felder leer gelassen.

Leider regelt der Adapter allerdings nicht. Beispiel: im Moment kommen ca. 4kW vom Dach, und der Charger ist auf die 6A Stufe eingestellt. Entsprechend lädt er mit 1392W (einphasig). Hausverbrauch sind aktuell ca. 700W und gute 2000W laufen in die Einspeisung Richtung Netzbetreiber. Eigentlich sollte der Adapter doch jetzt den Charger dorthingehend anpassen, dass er die Ladeleistung auf 6A + 2000W (aus Überschuss) anpasst, oder? Tut er leider nicht :( Was mache ich falsch?

Hier mal ein Screenshot aus meiner VIS, in der ich die jeweiligen Werte mal schnell eingebaut habe (Werte haben sich inzwischen leicht geändert):

![Screenshot 2022-03-02 112100](https://user-images.githubusercontent.com/19758651/156343382-92fd76c8-e33e-412b-8b69-ff3fcdbeaa8a.jpg)

Danke und viele Grüße
Christian

Hi, first of all a big THANK YOU for the adapter and all the work that obviously went into it. Great!

My question would be: I understand the bottom parts in the adapter config in such a way that you enter the PV power and the household consumption there (I have kept to https://github.com/MK-2001/ioBroker.go-e/discussions/21#discussioncomment-888287). The data points I#ve entered contain valid values that are updated every 10sec. Since I don't have a battery, I left the lowest two fields empty.

Unfortunately, however, the adapter does not regulate. Example: at the moment there is approx. 4kW coming from the roof, and the charger is set to the 6A level. Accordingly, it charges with 1392W (single-phase). Household consumption is currently approx. 700W and a ~2000W is being fed into the grid. Actually, the adapter should now adjust the charger so that it adjusts the charging power to 6A + 2000W (from the PV value delta), right? Unfortunately it does not :( What am I doing wrong?

Thank you and best regards
Christian

**Comment by MK-2001** [comment]
Hi, 

kannst du genauer beschreiben, welche "eingetragenen Datenpunkte" du meinst? Schreibt du die Datenpunkte? 
Was ist in den Log ersichtlich? Vor allem wenn du auf debug stellst?

Gruß

**Comment by FChristian** [comment, marked as answer]
Hi, die Datenpunkte werden von anderen Adaptern bzw Skripten geschrieben. Die Datenquellen sind:
- Wert "Energie" (holt aus dem WR die aktuelle PV Leistung in Watt) -> habe ich bei "aktuell verfügbare Solarleistung" eingetragen
- Wert "Verbrauch" (holt aus dem Smartmeter den aktuellen Hausverbrauch) -> habe ich bei "aktuellen Stromverbrauch zu Hause" eingetragen

Debug habe ich jetzt aktiviert, sage Bescheid was rauskommt. Leider ist das Auto gerade weg, aber ich sehe schonmal dass die eingetragenen Werte offenbar ankommen:

go-e.1 | 2022-03-02 11:51:13.374 | debug | state javascript.0.Stromzaehler.Verbrauch changed: 781 (ack = true) namespace: go-e.1
go-e.1 | 2022-03-02 11:51:13.191 | debug | state javascript.0.Energie changed: 4357 (ack = true) namespace: go-e.1

**Reply by MK-2001** [reply]
Btw. ich habe gerade im code geschaut... Wenn das Attribute als ack = true ankommt, dann wird es nicht verarbeitet, da der Adapter denkt, dass dieses bereits verarbeitet wurde. 

Nun kann man darüber diskutieren, ob das auch für fremde Attribute gelten darf, oder nicht... Dennoch damit der Adapter in der Version 1.0.21 funktioniert muss es auf ack = false stehen.

**Reply by FChristian** [reply]
Hm das wird schwierig, da noch andere Adapter darauf zugreifen.. habe auch gerade mal aus Spass ein paar andere Adapter und Datenpunkte angeschaut, bei denen Werte direkt von den Adaptern geschrieben werden, dort steht überall Bestätigt=Wahr... ich wüsste ehrlich gesagt nicht mal, wie man die auf "nicht bestätigt" umsetzen sollte..

**Reply by MK-2001** [reply]
Opened Feature #125 

Bei mir und anderen Adaptern wird der Wert beim setzen als ack = false geschrieben. Wenn dieser dann an die HW weitergegeben wurde, wird dieser dan nochmals als ack=true geschrieben. 

Das macht man bei Javascript Elementen meist keinen Sinn, da keine Hardware dahinter steckt.

**Comment by MK-2001** [comment]
Please recheck.
